src/Antonova/Task5/Tracer.cpp

```diff
@@ -0,0 +1,89 @@
+#include "Tracer.h"
+#include "QtMath"
+
+QVector3D Tracer::mix(const QVector3D &a, const QVector3D &b, const float k) {
+  return a * k + b * (1 - k);
+}
+
+QVector3D Tracer::reflect(const QVector3D &d, const QVector3D &n) {
+  return d - 2.f * n * QVector3D::dotProduct(d, n);
+}
+
+QVector3D Tracer::refract(const QVector3D &incident, const QVector3D &normal,
+                          const float n_1, const float n_2 = 1.f) {
+  auto cos_1 =
```

Comment:
Все что можно делать `const` нужно делать константным
```suggestion
  const auto cos_1 =
```